extensions/iidm/include/powsybl/iidm/extensions/iidm/VoltagePerReactivePowerControlAdder.hpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_EXTENSIONS_IIDM_VOLTAGEPERREACTIVEPOWERCONTROLADDER_HPP
+#define POWSYBL_IIDM_EXTENSIONS_IIDM_VOLTAGEPERREACTIVEPOWERCONTROLADDER_HPP
+
+#include <powsybl/iidm/ExtensionAdder.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace iidm {
+
+class VoltagePerReactivePowerControlAdder : public ExtensionAdder {
+public:
+    /**
+     * Constructor
+     */
+    explicit VoltagePerReactivePowerControlAdder(Extendable& extendable);
+
+    /**
+     * Destructor
+     */
+    ~VoltagePerReactivePowerControlAdder() noexcept override = default;
+
+    /**
+     * Copy constructor
+     */
+    VoltagePerReactivePowerControlAdder(const VoltagePerReactivePowerControlAdder&) = default;
+
+    /**
+     * Move constructor
+     */
+    VoltagePerReactivePowerControlAdder(VoltagePerReactivePowerControlAdder&&) = default;
+
+    /**
+     * Copy assignment operator
+     */
+    VoltagePerReactivePowerControlAdder& operator=(const VoltagePerReactivePowerControlAdder&) = delete;
+
+    /**
+     * Move assignment operator
+     */
+    VoltagePerReactivePowerControlAdder& operator=(VoltagePerReactivePowerControlAdder&&) = delete;
+
+    /**
+     * Set slope value
+     *
+     * @param slope The slope value
+     *
+     * @return this VoltagePerReactivePowerControlAdder object
+     */
+    VoltagePerReactivePowerControlAdder& withSlope(double slope);
+
+protected:  // ExtensionAdder
+    /**
+     * Creates the VoltagePerReactivePowerControl extension.
+     *
+     * @param extendable the extendable
+     *
+     * @return the extension
+     */
+    std::unique_ptr<Extension> createExtension(Extendable& extendable) override;
+
+private:
+    double m_slope = 0.0;
```

Comment:
Why the default value is 0. Should it be nan?

Reply:
I agree, but it's Java behaviour. Do I create an issue ?

Reply:
Fixed by initializing to `stdcxx::nan()`

Reply:
Could you show me the code that initialize the slope to 0 in the java code? From my understanding, the value is not initialized in the adder, so it is set to NaN.

Reply:
It sounds that 0.0d is the default value for double : https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html
But maybe I misunderstood.

Reply:
Hum... you right... my bad. I prefer revert this fix, and be consistent with the java code even I think it's a mistake.